src/Storages/StorageSnapshot.h

```diff
@@ -0,0 +1,86 @@
+#pragma once
+#include <Storages/StorageInMemoryMetadata.h>
+
+namespace DB
+{
+
+class IStorage;
+
+/// Snapshot of storage that fixes set columns that can be read in query.
+/// There are 3 sources of columns: regular columns from metadata,
+/// dynamic columns from object Types, virtual columns.
+struct StorageSnapshot
```

Comment:
Hi @CurtizJ , could you kindly explain the difference between StorageInMemoryMetadata vs StorageSnapshot? e.g., when to use StorageInMemoryMetadata::getColumns and when to use StorageSnapshot::getColumns? Thank you!